pygmt/tests/test_x2sys_cross.py

```diff
@@ -0,0 +1,176 @@
+        output = x2sys_cross(
+            tracks=["@tut_ship.xyz"], tag=tag, coe="i", outfile=outfile, verbose="i"
+        )
+
+        assert output is None  # check that output is None since outfile is set
+        assert os.path.exists(path=outfile)  # check that outfile exists at path
+        _ = pd.read_csv(outfile, sep="\t", header=2)  # ensure ASCII text file loads ok
+
+    return output
+
+
+def test_x2sys_cross_input_file_output_dataframe(mock_x2sys_home):
+    """
+    Run x2sys_cross by passing in a filename, and output internal crossovers to
+    a pandas.DataFrame
+    """
+    with TemporaryDirectory(prefix="X2SYS", dir=os.getcwd()) as tmpdir:
+        tag = os.path.basename(tmpdir)
+        x2sys_init(tag=tag, fmtfile="xyz", force=True)
+        output = x2sys_cross(tracks=["@tut_ship.xyz"], tag=tag, coe="i", verbose="i")
+
+        assert isinstance(output, pd.DataFrame)
+        assert output.shape == (14294, 12)
+        columns = list(output.columns)
+        assert columns[:6] == ["x", "y", "i_1", "i_2", "dist_1", "dist_2"]
+        assert columns[6:] == ["head_1", "head_2", "vel_1", "vel_2", "z_X", "z_M"]
+
+    return output
+
+
+def test_x2sys_cross_input_dataframe_output_dataframe(mock_x2sys_home, tracks):
+    """
+    Run x2sys_cross by passing in one dataframe, and output external crossovers
+    to a pandas.DataFrame. Not actually implemented yet, wait for
+    https://github.com/GenericMappingTools/gmt/issues/3717
+    """
+    with TemporaryDirectory(prefix="X2SYS", dir=os.getcwd()) as tmpdir:
+        tag = os.path.basename(tmpdir)
+        x2sys_init(tag=tag, fmtfile="xyz", force=True)
+
+        with pytest.raises(NotImplementedError):
+            _ = x2sys_cross(tracks=tracks, tag=tag, coe="i", verbose="i")
+
+        # assert isinstance(output, pd.DataFrame)
+        # assert output.shape == (4, 12)
+        # columns = list(output.columns)
+        # assert columns[:6] == ["x", "y", "t_1", "t_2", "dist_1", "dist_2"]
+        # assert columns[6:] == ["head_1","head_2","vel_1","vel_2","z_X","z_M"]
+        # assert output.dtypes["t_1"].type == np.datetime64
+        # assert output.dtypes["t_2"].type == np.datetime64
+
+    # return output
+
+
+def test_x2sys_cross_input_two_filenames(mock_x2sys_home):
+    """
+    Run x2sys_cross by passing in two filenames, and output external crossovers
+    to a pandas.DataFrame
+    """
+    with TemporaryDirectory(prefix="X2SYS", dir=os.getcwd()) as tmpdir:
+        tag = os.path.basename(tmpdir)
+        x2sys_init(tag=tag, fmtfile="xyz", force=True)
+
+        # Create temporary xyz files
+        for i in range(2):
+            np.random.seed(seed=i)
+            with open(os.path.join(os.getcwd(), f"track_{i}.xyz"), mode="w") as fname:
+                np.savetxt(fname=fname, X=np.random.rand(10, 3))
+
+        output = x2sys_cross(
+            tracks=["track_0.xyz", "track_1.xyz"], tag=tag, coe="e", verbose="i"
+        )
```

Comment:
This test produces two files `track_0.xyz` and `track_1.xyz` in the `tmp-test-dir-with-unique-name` directory. 

I'm not fully understand what the test does. It should either output to a temporary directory, or delete these two files at the end of the test.